client/src/pages/ChatsBlock.jsx

```diff
@@ -0,0 +1,297 @@
+      </div>
+
+      {anyError && <p style={{ color: 'red', marginTop: '5px' }}>{anyError}</p>}
+
+      <div className='border-solid w-full h-70vh p-3 overflow-y-scroll'>
+        {allChats.map((chat) => {
+          return (
+            <div
+              key={chat.id}
+              className='px-3 h-10 rounded-xl flex justify-between items-center mb-2 bg-slate-400'
+            >
+              <div className='flex items-center'>
+                <p className='font-bold text-xl mr-4'>{chat.name}</p>
+                {chat.creator && (
+                  <p className='text-gray-500'>{`created by "${chat.creator}"`}</p>
+                )}
+              </div>
+
+              <div className='flex h-8'>
+                {chat.creator && (
+                  <Formik
+                    initialValues={{ renameValue: '' }}
+                    validateOnMount={true}
+                    onSubmit={({ renameValue }, formikHelpers) => {
+                      formikHelpers.setSubmitting(true);
+                      formikHelpers.resetForm();
+
+                      userService
+                        .renameRoom(renameValue, chat.id)
+                        .then((res) => {
+                          setAllChats(res);
+                        })
+                        .catch((error) => {
+                          if (error.message) {
+                            setAnyError(error.message);
+                          }
+
+                          if (!error.response?.data) {
+                            return;
+                          }
+
+                          const { errors, message } = error.response.data;
+
+                          formikHelpers.setFieldError(
+                            'renameValue',
+                            errors?.renameValue
+                          );
+
+                          if (message) {
+                            setAnyError(message);
+                          }
+                        })
+                        .finally(() => formikHelpers.setSubmitting(false));
+                    }}
+                  >
+                    {({ touched, errors, isSubmitting, resetForm, values }) => (
+                      <Form className='flex items-center h-8'>
+                        <div className='relative'>
+                          <div>
+                            <Field
+                              className='h-full rounded-xl p-2 w-40'
+                              validate={chatNameValidation}
+                              name='renameValue'
+                              type='text'
+                              id='renameValue'
+                              placeholder='new name for the chat'
+                              disabled={!user}
+                            />
+                          </div>
+
+                          {touched.renameValue &&
+                            errors.renameValue &&
+                            setAnyError(errors.renameValue)}
```

Comment:
Directly calling `setAnyError(errors.renameValue)` inside the JSX can lead to unnecessary re-renders. Consider handling this logic inside a `useEffect` or a separate function.